### PR TITLE
Small Maven improvements

### PR DESCRIPTION
**Pull request by arend-von-reinersdorff**
## Overview

- Remove default jar packaging
- Unify indent with tabs in one location
- Remove unnecessary compiler plugin

---
I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: junit5-jupiter-starter-maven/pom.xml [hunk 1]
```diff
@@ -35,10 +35,6 @@
 
 	<build>
 		<plugins>
-			<plugin>
-				<artifactId>maven-compiler-plugin</artifactId>
-				<version>3.8.1</version>
-			</plugin>
```
Comment (marcphilipp) [inline junit5-jupiter-starter-maven/pom.xml:41]: ❌ This causes Maven to use a more recent version of the maven-compiler-plugin than declared in its super POM. AFAIK that's a "best practice" and should be kept.

**Review by marcphilipp — CHANGES_REQUESTED** [review]

File: junit5-jupiter-starter-maven/pom.xml (hunk 1 above)
Comment (arend-von-reinersdorff) [reply junit5-jupiter-starter-maven/pom.xml:41]: @marcphilipp I agree in general, that explicitly overriding plugin versions from the super POM can be a good practice.

But I think the sample projects should be kept small and only use configuration actually needed for Junit 5. Currently, the Maven starter declares two plugins:
- The maven-surefire-plugin which is essential for Junit 5. Without this declaration, the tests won't run
- The maven-compiler-plugin which is not needed for Junit 5

I think a casual reader will not understand the difference. That's why I propose to remove the declaration of the maven-compiler-plugin.

Comment (marcphilipp) [reply junit5-jupiter-starter-maven/pom.xml:41]: @sormuras WDYT?

Comment (sormuras) [reply junit5-jupiter-starter-maven/pom.xml:41]: As long as the sample projects to compile and test successfully, I'm fine with pruning not (longer) required configuration details. IIRC, some of the samples failed to build when using the default m-c-p ... but that might be outdated (and depend on the version of Maven being used).

**Review by sormuras — CHANGES_REQUESTED** [review]
There are two more `pom.xml` in this repository that should be also improved like the other two already touched by this PR.

image

**Review by sormuras — APPROVED** [review]

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
@arend-von-reinersdorff Thanks! 👍

**Pull request merged**